scripts/quests/jeuno/LB09_1_Dormant_Powers_Dislodged.lua

```diff
@@ -0,0 +1,151 @@
+    [7] = { xi.items.DANGRUF_STONE            },
+    [8] = { xi.items.ORCISH_AXE               },
+    [9] = { xi.items.QUADAV_BACKSCALE         },
+    -- 10+ = It repeats the pattern ad nauseam. Supposedly, it can request more items, but I cannot figure them out without a cap.
+}
+
+quest.reward =
+{
+    fame  = 50,
+    fameArea = JEUNO,
+    keyItem = xi.ki.SOUL_GEM,
+}
+
+quest.sections =
+{
+    -- Section: Quest available.
+    {
+        check = function(player, status)
+            return status == QUEST_AVAILABLE and
+                player:getMainLvl() >= 86 and
+                player:getLevelCap() == 90 and
+                xi.settings.MAX_LEVEL >= 95
+        end,
+
+        [xi.zone.RULUDE_GARDENS] =
+        {
+            ['Nomad_Moogle'] =
+            {
+                onTrigger = function(player, npc)
+                    -- Requested item is an additional argument. (csid, 0, 1, 4, 0, 0, ITEM). See table on top.
+                    local itemWanted = math.random(0, 9)
+                    quest:setVar(player, 'itemWanted', itemWanted + 1)
+                    return quest:progressEvent(10045, 0, 1, 4, 0, 0, itemWanted)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [10045] = function(player, csid, option, npc)
+                    if option == 11 then -- Accept quest option.
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.RULUDE_GARDENS] =
+        {
+            ['Nomad_Moogle'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(10192) -- Timing Minigame starting event.
+                    else
+                        local itemWanted = quest:getVar(player, 'itemWanted') - 1
+                        return quest:event(10045, 0, 1, 4, 1, 0, itemWanted)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local itemWanted  = quest:getVar(player, 'itemWanted') - 1
+                    local itemToTrade = itemWantedTable[itemWanted][1]
+
+                    if npcUtil.tradeHasExactly(trade, {{xi.items.KINDREDS_CREST, 1}, {itemToTrade, 1}}) and
+                        player:getMeritCount() > 9
+                    then
+                        return quest:progressEvent(10191)
+                    end
+                end,
+            },
+
+            onEventUpdate =
+            {
+                [10192] = function(player, csid, option, npc)
+                    -- Option: Push! The option is the number of frames. 1 sec = 60 frames. Probably.
+                    if option >= 901 then
```

Comment:
These options smell like a bitmask, and might be able to be broken out

Reply:
I dont really know, but it seemed like a progresive count.
The moment the option is selected in client, the option ID seems to be set to the time passed since the start. (Not the start of the event, the start of the counting.)